SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/Storage.cs

```diff
@@ -343,14 +343,40 @@ private static bool ChannelMatchesIdentifier(string channelName, string identifi
 
         private Task AddTaskToQueue(Action action)
         {
-            var task = new Task(action);
+            var task = new Task(() =>
+            {
+                try
+                {
+                    action();
+                }
+                catch (Exception e) when (e.GetType() != typeof(StorageException))
```

Comment:
Not sure why we need exact type since our other catch blocks by definition catch sub-classes.

How about `!(e is StorageException)` instead?